python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt/sling/asset_defs.py

```diff
@@ -123,11 +123,9 @@ def build_assets_from_sling_stream(
     Args:
         source (SlingConnectionResource): The source SlingConnectionResource to use.
         target (SlingConnectionResource): The target SlingConnectionResource to use.
-        stream (str): The source stream to sync from. This can be a table, a query, or a path.
+        streams (str): The streams to sync from and any configurations specific to the stream. This can be a table, a query, or a path. See the Sling documentation for more information. https://docs.slingdata.io/sling-cli/replication
         target_object (str, optional): The target object to sync to. This can be a table, or a path. Defaults to the template of "{{target_schema}}.{{stream_schema}}_{{stream_table}}". See the Sling documentation for more information. https://docs.slingdata.io/sling-cli/replication
         mode (SlingMode, optional): The sync mode to use when syncing. Defaults to `full-refresh`.
-        primary_key (Optional[Union[str, List[str]]], optional): The optional primary key to use when syncing.
```

Comment:
I pushed writing primary and update keys down to each individual stream because the API/docs weren't clear if those were valid defaults at the top-level.

ex.
```python
asset_def_update = build_assets_from_sling_streams(
    sling_file_connection,
    sling_sqlite_connection,
    asset_specs,
    streams=[
        {
            "stream_name": f"file://{test_csv}",
            "primary_key": "SPECIES_NAME",
            "update_key": "UPDATED_AT",
        }
    ],
    target_object="main.tbl",
    mode=SlingMode.INCREMENTAL,
)

```

Since the docs were written and the `Configuration` page was flushed out some more, it looks every config for a stream can be put into the `defaults` now. Might be worth making a `default_stream_configs` parameter that `primary_key` and `update_key` can also reside in.

If that's jumping the gun too much, then the high-level question is: what expectations are there for the API for setting stream-level configs, ex. default+granular configs, typing, flexibility as Sling adds more configs, etc.

Let's talk at our call later whether we want to support both top-level and granular configs, or just the latter.